frontend/packages/console-shared/src/components/dropdown/dropdown-with-switch/DropdownWithSwitchToggle.tsx

```diff
@@ -0,0 +1,82 @@
+import * as React from 'react';
+import { MenuToggle, Popper } from '@patternfly/react-core';
+
+const DropdownWithSwitchToggle: React.FC<DropdownWithSwitchToggleProps> = ({
+  isDisabled,
+  isFullWidth,
+  isOpen,
+  label,
+  menu,
+  menuRef,
+  onToggle,
+}) => {
+  const toggleRef = React.useRef(null);
+  const containerRef = React.useRef(null);
+  const handleMenuKeys = (event) => {
+    if (menuRef.current) {
+      if (event.key === 'Escape') {
+        onToggle(false);
+        toggleRef.current.focus();
+      }
+      if (!menuRef.current?.contains(event.target) && event.key === 'Tab') {
+        onToggle(false);
+      }
+    }
+  };
+  const handleMenuClick = (event) => {
```

Comment:
I think these `event` params need a type -- but the types we use are going to need more logic in your if-statement to get around TS showing DOM inconsistencies as errors. But I want to test these -- so I'm going to setup something to see if I can execute these handlers and then test out my proposed solution. Will report back.

Reply:
Whoops.  I missed the `KeyboardEvent`.  Will fix.

Reply:
Hmm.  TypeScript doesn't like:
```
    const handleMenuKeys = (event: KeyboardEvent) => {
      if (event.key === 'Tab' && !menuRef.current?.contains(event.target)) {
        onToggle(false);
      }
    };
```
as it produces this error:
```
Argument of type 'EventTarget' is not assignable to parameter of type 'Node'.
  Type 'EventTarget' is missing the following properties from type 'Node': baseURI, childNodes, firstChild, isConnected, and 43 more.
```